### PR TITLE
Changed the for loop in slave.js and master.js

### PR DESCRIPTION
**Pull request by rohilsurana**
Changed the for....in loop with normal for loop in slave.js and master.js. Some more instances left in master.js.

### CODE REVIEW

**Comment by ngsankha** [comment]
Any reason why you are iterating from the last value to the first? The traditional way is to go from `0` to `array.length` I believe?

Also the changes look good, convert the more instances of it in `master.js` and we will be good to go! :smiley:

File: master.js [hunk 1]
```diff
@@ -7,8 +7,8 @@
 
   var enlistServers = function() {
     slavesIp = JSON.parse(fs.readFileSync(test ? './tests/slaves.json' : 'slaves.json'));
-    for (var name in slavesIp) {
-      slavesLoad[name] = 0;
+    for (var i = slavesIp.length - 1; i >= 0; i -= 1) {
```
Comment (ngsankha) [inline master.js]: Also you can replace the `i -= 1` with `i--` or `i++` if you decide to loop from start to end.

**Comment by rohilsurana** [comment]
For i-- I read that I -= 1 seems more readable. And for the array.length thing , I did so to save a variable.
for(var i = 0,j = array.length; i < j; i -= 1) uses another variable

**Comment by ngsankha** [comment]
No, `i--` is just fine and concise for this case. So change to that.

Also you don't need a 2nd variable. You can always write:

``` javascript
for (var i = 0; i < array.length; i++)
```

**Comment by rohilsurana** [comment]
OK I'll use i-- but for the for loop when you are using that one you are iterating through the complete array each time you are using length i.e. each time loop runs

**Comment by ngsankha** [comment]
No. `array.length` is a constant that is stored for ever array object. You never need to run through all the elements of an array to use `array.length`.

**Comment by rohilsurana** [comment]
Hmm but we need to derefrence the array each time

**Comment by ngsankha** [comment]
You are thinking in C/C++ kind of manner. JavaScript arrays not like that. They are just a value that's a part of the array object. So there is no dereferencing or counting of the actual array elements whenever you access `array.length`.

Think of it as something like (though the actual implementation is much more complicated):

```
             ___________________________________________________
             | 8 | 'a' | 'b' | 'c' |.. 4 more elements ..| 'h' |
             |___|_____|_____|_____|_____________________|_____|
            length  0     1     2          .....            7
```

Thus when ever you say `array.length`, the value from the first slot is returned. When you say `array[0]` the second slot is returned and something like this happens.

**Comment by ngsankha** [comment]
This is awesome! Thanks! :smile:

**Pull request merged**